src/SignalRServiceExtension/Client/AzureSignalRClient.cs

```diff
@@ -4,88 +4,64 @@
 using System;
 using System.Collections.Generic;
 using System.IdentityModel.Tokens.Jwt;
-using System.Net.Http;
-using System.Net.Http.Headers;
 using System.Reflection;
 using System.Runtime.InteropServices;
 using System.Security.Claims;
 using System.Text;
 using System.Text.RegularExpressions;
 using System.Threading.Tasks;
+using Microsoft.Azure.SignalR.Management;
 using Microsoft.IdentityModel.Tokens;
-using Newtonsoft.Json;
 
 namespace Microsoft.Azure.WebJobs.Extensions.SignalRService
 {
     internal class AzureSignalRClient : IAzureSignalRSender
     {
-        private readonly HttpClient httpClient;
+        private readonly IServiceHubContextStore serviceHubContextStore;
+        private readonly IServiceManager serviceManager;
 
-        internal string BaseEndpoint { get; }
-        internal string AccessKey { get; }
-        internal string Version { get; set; }
-
-        internal AzureSignalRClient(string connectionString, HttpClient httpClient)
+        internal AzureSignalRClient(IServiceHubContextStore serviceHubContextStore, IServiceManager serviceManager)
         {
-            (BaseEndpoint, AccessKey, Version) = ParseConnectionString(connectionString);
-            this.httpClient = httpClient;
+            this.serviceHubContextStore = serviceHubContextStore;
+            this.serviceManager = serviceManager;
         }
 
-        internal SignalRConnectionInfo GetClientConnectionInfo(string hubName, IEnumerable<Claim> claims = null)
+        internal SignalRConnectionInfo GetClientConnectionInfo(string hubName, string userId)
         {
-            var hubUrl = $"{BaseEndpoint}/client/?hub={hubName}";
-            var identity = new ClaimsIdentity(claims);
-            var token = GenerateJwtBearer(null, hubUrl, identity, DateTime.UtcNow.AddMinutes(30), AccessKey);
             return new SignalRConnectionInfo
             {
-                Url = hubUrl,
-                AccessToken = token
+                Url = serviceManager.GetClientEndpoint(hubName),
+                AccessToken = serviceManager.GenerateClientAccessToken(hubName, userId)
             };
         }
 
-        internal SignalRConnectionInfo GetServerConnectionInfo(string hubName, string additionalPath = "")
+        public async Task SendToAll(string hubName, SignalRData data)
         {
-            var hubUrl = $"{BaseEndpoint}/api/v1/hubs/{hubName}";
-            var audienceUrl = $"{hubUrl}{additionalPath}";
-            var token = GenerateJwtBearer(null, audienceUrl, null, DateTime.UtcNow.AddMinutes(30), AccessKey);
-            return new SignalRConnectionInfo
-            {
-                Url = audienceUrl,
-                AccessToken = token
-            };
+            var serviceHubContext = await serviceHubContextStore.GetOrAddAsync(hubName);
+            await serviceHubContext.Clients.All.SendCoreAsync(data.Target, data.Arguments);
```

Comment:
>SendCoreAsync [](start = 48, length = 13)

When migration to ManagementSDK, make sure to add ProductInfo correctly for the request.

Reply:
If ServiceTransportType is Transient, the ProductInfo can be added to header,
but if the ServiceTransportType is Persistent, messages will be sent in websockets connection. There seems no way to declare `Asrs-User-Agent` like REST API. 
> request.Headers.Add("Asrs-User-Agent", GetProductInfo())

Is it OK that only Transient mode adds the Product Info?

Reply:
When use websockets connection we doesn't log message send requests. But could it be added when setup connection between functionapp and signalr service similar to the behavior between normal app server and signalr service. So it could be tracked.

Reply:
I reuse the signalr SDK, therefore, in Persistent mode, the product info is already added when starting connection to signalr service, but the package name will be `management SDK` since signalr sdk calls `Assembly.GetCallingAssembly()` to obtain package name. 
Although we can change the signalr sdk's interface to pass the assembly name, it is not elegant...

Reply:
Yes. Since Persistent mode reuse service sdk to start connection, the production info is already in request header. But the problem is the package name will be Management SDK instead of . I need to change the api of server-connection-start-method.

Reply:
Will be add in management sdk